obspy/io/mseed/util.py

```diff
@@ -238,31 +275,110 @@ def get_timing_and_data_quality(file_or_file_object):
+        elif starttime is not None:
+            if this_info['starttime'] >= starttime:
+                if(io_flags):
+                    # Add the value of each bit to the io_count.
+                    for _i in range(6):
+                        if (this_info['io_and_clock_flags'] & (1 << _i)) != 0:
+                            io_count[_i] += 1
+                if(activity_flags):
+                    # Add the value of each bit to the activity_count.
+                    for _i in range(7):
+                        if (this_info['activity_flags'] & (1 << _i)) != 0:
+                            activity_count[_i] += 1
+                # Add the timing quality.
+                if 'timing_quality' in this_info:
+                    timing_quality.append(float(this_info['timing_quality']))
+                # Add the value of each bit to the quality_count.
+                for _i in range(8):
+                        if (this_info['data_quality_flags'] & (1 << _i)) != 0:
+                            quality_count[_i] += 1
+        elif endtime is not None:
+            if this_info['starttime'] <= endtime:
+                if(io_flags):
+                    # Add the value of each bit to the io_count.
+                    for _i in range(6):
+                        if (this_info['io_and_clock_flags'] & (1 << _i)) != 0:
+                            io_count[_i] += 1
+                if(activity_flags):
+                    # Add the value of each bit to the activity_count.
+                    for _i in range(7):
+                        if (this_info['activity_flags'] & (1 << _i)) != 0:
+                            activity_count[_i] += 1
+                # Add the timing quality.
+                if 'timing_quality' in this_info:
+                    timing_quality.append(float(this_info['timing_quality']))
+                # Add the value of each bit to the quality_count.
+                for _i in range(8):
+                        if (this_info['data_quality_flags'] & (1 << _i)) != 0:
+                            quality_count[_i] += 1
+        else:
+            if(io_flags):
+                # Add the value of each bit to the io_count.
+                for _i in range(6):
+                    if (this_info['io_and_clock_flags'] & (1 << _i)) != 0:
+                        io_count[_i] += 1
+            if(activity_flags):
+                # Add the value of each bit to the activity_count.
+                for _i in range(7):
+                    if (this_info['activity_flags'] & (1 << _i)) != 0:
+                        activity_count[_i] += 1
+            # Add the timing quality.
+            if 'timing_quality' in this_info:
+                timing_quality.append(float(this_info['timing_quality']))
+            # Add the value of each bit to the quality_count.
+            for _i in range(8):
+                    if (this_info['data_quality_flags'] & (1 << _i)) != 0:
+                        quality_count[_i] += 1
```

Comment:
The point is that when starttime and endtime are None (default) you should still be able to extract the flags. I didn't want to disrupt the current behaviour.
Starttime and endtime should be optional parameters that users specifie if they want to extract only subsets of the flags related to specific portions of the file(s)